### DISCUSSION TITLE
File names not displaying on S3 mounts

### DISCUSSION THREAD

**Discussion by melmatsuoka** [iterate-ch/cyberduck#16299] [Q&A]
I'm having an issue with file names not showing up on S3 mounts, using Mountain Duck 4.16.3 on macOS 14.6.1. The folder names all show up fine, but all the files within the folder don't show up.

The file names on the same S3 share, using the same S3 credentials, show up fine in Cyberduck, however.

![CleanShot%202024-09-09%20at%2009 51 40](https://github.com/user-attachments/assets/555af79b-8a0a-415d-8cc8-0692b55fdd3e)

**Comment by dkocher** [comment]
The obvious reason would be the filename matching the following regular expression which by defaults filters filenames in Mountain Duck.

```
fs.filenames.filter.name.regexp=[\\.]{1,2} .*[\\/\\\\].* \\.file-segments
```

However your screenshot does not indicate a match. Can you [enable debug logging](https://docs.mountainduck.io/mountainduck/support/#logging-output) and search the log file for lines containing `FilesystemListFilter`.

**Reply by melmatsuoka** [reply]
Not sure this is an issue with the regex, because literally *all* filenames (and not folder/directory names) are not appearing on my S3 mounts, across multiple different S3 buckets. 

Filenames show up fine on my Backblaze B2 and Google Drive mounts, however.

Here's a (truncated) log containing all lines containing `FilesystemListFilter` in my `mountainduck.log` file

**Reply by dkocher** [reply]
The log shows the files are filtered because of the applied [storage class](https://aws.amazon.com/s3/storage-classes/). The default in Mountain Duck is to skip all files in _Glacier_ or _Deep Archive_ storage classes, as these files cannot be retrieved without restoring first. 

The default setting is 

```
fs.filenames.filter.storageclass.regexp=GLACIER DEEP_ARCHIVE
```

**Reply by melmatsuoka** [reply]
hmm..apologies. Had to reupload the log file.

[mountainduck.log](https://github.com/user-attachments/files/17013740/mountainduck.log)

**Reply by melmatsuoka** [reply]
> The log shows the files are filtered because of the applied [storage class](https://aws.amazon.com/s3/storage-classes/). The default in Mountain Duck is to skip all files in _Glacier_ or _Deep Archive_ storage classes, as these files cannot be retrieved without restoring first.
> 
> The default setting is
> 
> ```
> fs.filenames.filter.storageclass.regexp=GLACIER DEEP_ARCHIVE
> ```

Ah, that's interesting. So you can't retrieve just the filenames of objects in Glacier/GDA without restoring them first? I dont have any issues with Cyberduck being able to do this with the default settings on the same S3 bucket(s)?

**Reply by dkocher** [reply]
Retrieving the keys (filenames) is not a problem but In Mountain Duck mounting as a filesystem it is very common that some application will attempt to read the file contents (for example to render a preview in Finder.app or Windows Explorer). Because that will fail for archived files we choose to not display them by default.

**Reply by melmatsuoka** [reply]
Ok that makes sense. Is there no practical workaround for this, at least on macOS? 

I'm using [NeoFinder](https://www.cdfinder.de) to index the files/folders in  my cloud archives (one of the new features recently added to NeoFinder was [direct support for indexing Mountain Duck mounted](https://www.cdfinder.de/guide/3/3.20/catalog_cloud_services.html) cloud volumes). I'm hoping there might be some way in situations like this to just display the names of the files (and any associated metadata such as file size, mod/create dates etc) without allowing system processes to actually read the contents of the files themselves if they specifically belong to the S3 Glacier/GDA storage class ?
